src/lib/autoinstall/ask/profile_reader.rb

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "autoinstall/ask/dialog"
+require "autoinstall/ask/question"
+require "autoinstall/ask/question_option"
+require "autoinstall/script"
+
+module Y2Autoinstall
+  module Ask
+    # This class turns an <ask-list> section into a list of Ask::Dialog objects
+    class ProfileReader
+      # Constructor
+      #
+      # @param ask_list_section [AskListSection] <ask-list> section from the profiel
+      # @param stage [Symbol] :initial or :cont
+      def initialize(ask_list_section, stage: :initial)
+        @ask_list_section = ask_list_section
+        @stage = stage
+      end
+
+      # Returns the dialogs according to the given set of <ask> sections
+      #
+      # @return [Array<Dialog>] List of dialogs
+      def dialogs
+        sorted_entries = ask_list_section
+          .entries
+          .select { |s| (s.stage&.to_sym || :initial) == stage }
```

Comment:
I would probably somehow make more visible that default if first stage if not defined. maybe constant with proper name?

Reply:
I found that default value for stage is not documented at https://documentation.suse.com/sles/15-SP2/single-html/SLES-autoyast/#CreateProfile-Ask

Reply:
> I found that default value for stage is not documented at https://documentation.suse.com/sles/15-SP2/single-html/SLES-autoyast/#CreateProfile-Ask

Yes, it is, below the example. Perhaps it is not the most prominent place.

Reply:
I overlook it. Others also have it at this place, so it is consistent. BTW why stage has different formatting? It looks strange.

Reply:
Yes, it is because it is missing the `<literal>` element. I am preparing a PR to fix that formatting and the `file name` element name too (it should be `filename`).